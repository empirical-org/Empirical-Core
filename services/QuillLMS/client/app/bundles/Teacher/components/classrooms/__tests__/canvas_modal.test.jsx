import React from 'react';
import { render } from '@testing-library/react';

import CanvasModal from '../canvas_modal'; // Adjust the import path as necessary

const TEACHER = 'teacher'
const ADMIN = 'admin'
const TEACHER_PREMIUM = 'Teacher Premium'
const SCHOOL_PREMIUM = 'School Premium'

describe('CanvasModal', () => {
  const mockClose = jest.fn();

  const renderWithProps = (userProps) => {
    return render(<CanvasModal close={mockClose} user={userProps} />);
  };

  it('renders correctly for TEACHER_LACKS_RELEVANT_PREMIUM', () => {
    const userProps = {
      role: TEACHER,
      subscription: { account_type: TEACHER_PREMIUM },
      school_linked_to_canvas: false
    };

    const { asFragment } = renderWithProps(userProps);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly for TEACHER_LACKS_SCHOOL_CANVAS', () => {
    const userProps = {
      role: TEACHER,
      subscription: { account_type: SCHOOL_PREMIUM },
      school_linked_to_canvas: false
    };

    const { asFragment } = renderWithProps(userProps);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly for TEACHER_LACKS_INDIVIDUAL_CANVAS', () => {
    const userProps = {
      role: TEACHER,
      subscription: { account_type: SCHOOL_PREMIUM },
      school_linked_to_canvas: true
    };

    const { asFragment } = renderWithProps(userProps);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly for ADMIN_LACKS_RELEVANT_PREMIUM', () => {
    const userProps = {
      role: ADMIN,
      subscription: { account_type: TEACHER_PREMIUM },
      school_linked_to_canvas: false
    };

    const { asFragment } = renderWithProps(userProps);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly for ADMIN_LACKS_SCHOOL_CANVAS', () => {
    const userProps = {
      role: ADMIN,
      subscription: { account_type: SCHOOL_PREMIUM },
      school_linked_to_canvas: false
    };

    const { asFragment } = renderWithProps(userProps);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly for ADMIN_LACKS_INDIVIDUAL_CANVAS', () => {
    const userProps = {
      role: ADMIN,
      subscription: { account_type: SCHOOL_PREMIUM },
      school_linked_to_canvas: true
    };

    const { asFragment } = renderWithProps(userProps);
    expect(asFragment()).toMatchSnapshot();
  });

});
