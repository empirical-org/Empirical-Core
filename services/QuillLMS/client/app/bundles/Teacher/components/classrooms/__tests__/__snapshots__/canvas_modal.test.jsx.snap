// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasModal renders correctly for ADMIN_LACKS_INDIVIDUAL_CANVAS 1`] = `
<DocumentFragment>
  <div
    class="modal-container canvas-modal-container"
  >
    <div
      class="modal-background"
    />
    <div
      class="canvas-modal quill-modal modal-body"
    >
      <div>
        <h3
          class="title"
        >
          You’re one step away from using Canvas
        </h3>
      </div>
      <p>
        Looking to streamline your teaching process with Quill’s Canvas integration? Great news! Your subscription gives you access to the integration and it has been set up by you or another administrator for your school. The last step is for you to log in to Quill via your Canvas account. Once you have done so, you’ll be able to:
      </p>
      <ul>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Seamlessly import your Canvas rosters.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Automatically create and sync Canvas student accounts.
        </li>
      </ul>
      <div
        class="form-buttons"
      >
        <button
          class="quill-button-archived outlined secondary medium focus-on-light"
          type="button"
        >
          Cancel
        </button>
        <div
          class="right-aligned-buttons"
        >
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="https://support.quill.org/en/articles/7995794-school-district-premium-how-can-i-import-classes-from-canvas"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn how
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CanvasModal renders correctly for ADMIN_LACKS_RELEVANT_PREMIUM 1`] = `
<DocumentFragment>
  <div
    class="modal-container canvas-modal-container"
  >
    <div
      class="modal-background"
    />
    <div
      class="canvas-modal quill-modal modal-body"
    >
      <div>
        <h3
          class="title"
        >
          Unlock Canvas with School or District Premium
        </h3>
      </div>
      <p>
        Looking to streamline your school’s teaching process with Quill’s Canvas integration? Subscribe to School or District Premium to allow teachers at your school or district to:
      </p>
      <ul>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Seamlessly import their canvas rosters.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Automatically create and sync Canvas student accounts.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Access a host of other premium benefits, including priority technical support, enhanced reporting, and assistance from our professional learning team.
        </li>
      </ul>
      <div
        class="form-buttons"
      >
        <button
          class="quill-button-archived outlined secondary medium focus-on-light"
          type="button"
        >
          Cancel
        </button>
        <div
          class="right-aligned-buttons"
        >
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="/premium"
            rel="noopener noreferrer"
            target="_blank"
          >
            Explore premium
          </a>
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="https://support.quill.org/en/articles/8337988-how-do-i-set-up-the-canvas-integration-for-my-school-district-for-canvas-quill-administrators"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn more
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CanvasModal renders correctly for ADMIN_LACKS_SCHOOL_CANVAS 1`] = `
<DocumentFragment>
  <div
    class="modal-container canvas-modal-container"
  >
    <div
      class="modal-background"
    />
    <div
      class="canvas-modal quill-modal modal-body"
    >
      <div>
        <h3
          class="title"
        >
          Get started with Canvas
        </h3>
      </div>
      <p>
        Looking to streamline your school’s teaching process with Quill’s Canvas integration? Great news! Your subscription gives you access to the integration. The next step is to set it up. This will allow teachers at your school or district to:
      </p>
      <ul>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Seamlessly import their canvas rosters.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Automatically create and sync Canvas student accounts.
        </li>
      </ul>
      <div
        class="form-buttons"
      >
        <button
          class="quill-button-archived outlined secondary medium focus-on-light"
          type="button"
        >
          Cancel
        </button>
        <div
          class="right-aligned-buttons"
        >
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="https://support.quill.org/en/articles/8337988-how-do-i-set-up-the-canvas-integration-for-my-school-district-for-canvas-quill-administrators"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn more
          </a>
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="/teachers/premium_hub/integrations"
            rel="noopener noreferrer"
            target="_blank"
          >
            Set up integration
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CanvasModal renders correctly for TEACHER_LACKS_INDIVIDUAL_CANVAS 1`] = `
<DocumentFragment>
  <div
    class="modal-container canvas-modal-container"
  >
    <div
      class="modal-background"
    />
    <div
      class="canvas-modal quill-modal modal-body"
    >
      <div>
        <h3
          class="title"
        >
          You’re one step away from using Canvas
        </h3>
      </div>
      <p>
        Looking to streamline your teaching process with Quill’s Canvas integration? Great news! Your subscription gives you access to the integration and it has been set up by an administrator for your school. The last step is for you to log in to Quill via your Canvas account. Once you have done so, you’ll be able to:
      </p>
      <ul>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Seamlessly import your Canvas rosters.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Automatically create and sync Canvas student accounts.
        </li>
      </ul>
      <div
        class="form-buttons"
      >
        <button
          class="quill-button-archived outlined secondary medium focus-on-light"
          type="button"
        >
          Cancel
        </button>
        <div
          class="right-aligned-buttons"
        >
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="https://support.quill.org/en/articles/7995794-school-district-premium-how-can-i-import-classes-from-canvas"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn how
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CanvasModal renders correctly for TEACHER_LACKS_RELEVANT_PREMIUM 1`] = `
<DocumentFragment>
  <div
    class="modal-container canvas-modal-container"
  >
    <div
      class="modal-background"
    />
    <div
      class="canvas-modal quill-modal modal-body"
    >
      <div>
        <h3
          class="title"
        >
          Unlock Canvas with School or District Premium
        </h3>
      </div>
      <p>
        Looking to streamline your teaching process with Quill’s Canvas integration? Subscribe to School or District Premium to:
      </p>
      <ul>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Seamlessly import your Canvas rosters.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Automatically create and sync Canvas student accounts.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Access a host of other premium benefits, including priority technical support, enhanced reporting, and assistance from our professional learning team.
        </li>
      </ul>
      <div
        class="form-buttons"
      >
        <button
          class="quill-button-archived outlined secondary medium focus-on-light"
          type="button"
        >
          Cancel
        </button>
        <div
          class="right-aligned-buttons"
        >
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="/premium"
            rel="noopener noreferrer"
            target="_blank"
          >
            Explore premium
          </a>
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="mailto:?subject=Can you help unlock Quill's Canvas integration?&body=Hi [Insert Your Administrator’s Name],%0D%0A%0D%0AI've been digging into the Canvas integration in Quill, and it seems like a fantastic tool for our teaching needs. To tap into its full potential, we'd need a School Premium or District Premium subscription.%0D%0A%0D%0AFor more details on Quill's premium offerings, they have a page listing the benefits and pricing here: https://www.quill.org/premium.%0D%0A%0D%0AIf you're curious about how Quill’s Canvas integration works, you can learn more here: https://support.quill.org/en/articles/8337988-how-do-i-set-up-the-canvas-integration-for-my-school-district-for-canvas-quill-administrators.%0D%0A%0D%0AI'd appreciate it if you would consider reaching out to Quill at sales@quill.org to discuss the possibility of upgrading our subscription. This could open up some exciting opportunities for us.%0D%0A%0D%0AThank you for your time and consideration.%0D%0A%0D%0ABest regards,%0D%0A%0D%0A[Insert Your Name]"
            rel="noopener noreferrer"
            target="_blank"
          >
            Contact your administrator
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CanvasModal renders correctly for TEACHER_LACKS_SCHOOL_CANVAS 1`] = `
<DocumentFragment>
  <div
    class="modal-container canvas-modal-container"
  >
    <div
      class="modal-background"
    />
    <div
      class="canvas-modal quill-modal modal-body"
    >
      <div>
        <h3
          class="title"
        >
          Contact your administrator to unlock Canvas
        </h3>
      </div>
      <p>
        Looking to streamline your teaching process with Quill’s Canvas integration? Great news! Your subscription gives you access to the integration. The next step is for your administrator to set it up. Once they have done so, you’ll be able to:
      </p>
      <ul>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Seamlessly import your Canvas rosters.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Automatically create and sync Canvas student accounts.
        </li>
      </ul>
      <div
        class="form-buttons"
      >
        <button
          class="quill-button-archived outlined secondary medium focus-on-light"
          type="button"
        >
          Cancel
        </button>
        <div
          class="right-aligned-buttons"
        >
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="https://support.quill.org/en/articles/7995794-school-district-premium-how-can-i-import-classes-from-canvas"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn more
          </a>
          <a
            class="quill-button-archived contained primary medium focus-on-light"
            href="mailto:?subject=Can you help set up Quill's Canvas integration?&body=Hi [Insert Your Administrator’s Name],%0D%0A%0D%0AI've been digging into the Canvas integration in Quill, and it seems like a fantastic tool for our teaching needs. The good news is that the integration is already unlocked because we have a subscription. I just need an administrator’s help to set it up.%0D%0A%0D%0AYou can view Quill’s instructions on how to set up the Canvas integration here: https://support.quill.org/en/articles/8337988-how-do-i-set-up-the-canvas-integration-for-my-school-district-for-canvas-quill-administrators%0D%0A%0D%0AThank you for your time and consideration.%0D%0A%0D%0ABest regards,%0D%0A%0D%0A[Insert Your Name]"
            rel="noopener noreferrer"
            target="_blank"
          >
            Contact your administrator
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
